controller/lib/hal/hal_stm32.cpp

```diff
@@ -311,6 +310,11 @@ void HalApi::startLoopTimer(const Duration &period, void (*callback)(void *),
   controller_callback = callback;
   controller_arg = arg;
 
+  // Init the watchdog timer now.  The watchdog timer is serviced by the
+  // loop callback function.  I don't init it until the loop starts because
+  // otherwise it may expire before the function that resets it starts running
+  watchdog_init();
```

Comment:
What if the chip hangs during setup?

Reply:
Yes, I agree, this isn't really the place to start the watchdog timer.

I think we should leave it here for now and I'll file an issue so we remember to move it after the covent deadline.  The proper place to initialize it is in the boot loader before our code even starts, but we don't even have a boot loader yet.  I'm sure we'll need one, but it's not something that needs to be done immediately.

The reason I think this is OK for now is that the ventilator can't be in use before the watchdog timer is currently setup to start so if the system hangs then it will hang just a few seconds after power-up and before it's actually being used.  That will be annoying, but not life threatening.